src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> previousNode = tail;
+        Node<T> newNode = new Node<>(previousNode, value, null);
+        tail = newNode;
+        if (previousNode == null) {
+            head = newNode;
+        } else {
+            previousNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        Node<T> newNode = new Node<>(node.prev, value, node);
+        if (node.prev != null) {
+            node.prev.next = newNode;
+        } else {
+            head = newNode;
+        }
+        node.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        T oldNodeValue = node.value;
+        node.value = value;
+        return oldNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        unlink(node);
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> node = head; node != null; node = node.next) {
+            if (node.value == null && object == null
+                    || node.value != null && node.value.equals(object)) {
```

Comment:
```suggestion
            if (node.value == object || node.value != null && node.value.equals(object)) {
```

Reply:
Fixed. IDEA cast expression to Objects.equals(node.value, object) , so I change your variant to IDEA variant.